selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsProjectsTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.dashboard.workspaces.details;
+
+import static org.eclipse.che.selenium.core.project.ProjectTemplates.MAVEN_SPRING;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjects.BottomButton.APPLY_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjects.BottomButton.SAVE_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.BottomButton.ADD_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.BottomButton.CANCEL_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples.UpperButton.SAMPLES_BUTTON;
+
+import com.google.inject.Inject;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.che.commons.lang.NameGenerator;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjects;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceProjectsSamples;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.Workspaces;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class WorkspaceDetailsProjectsTest {
+  private static final String PROJECT_NAME = NameGenerator.generate("wsDetails", 4);
+  private static final String PROJECT_FOR_SEARCHING_NAME = NameGenerator.generate("searchWs", 4);
+  private static final String SPRING_SAMPLE_NAME = "web-java-spring";
+  private static final String CONSOLE_SAMPLE_NAME = "console-java-simple";
+  private static final String EXPECTED_SUCCESS_NOTIFICATION = "Success\n" + "Workspace updated.";
+  private static final String CHECKING_CHECKBOXES_PROJECT_NAME = "checkboxesProject";
+  private static final String CHECKING_CHECKBOXES_NEVEST_PROJECT_NAME = "newestProject";
+  private static final String PROJECTS_TAB_NAME = "Projects";
+  private static final String EXPECTED_DELETE_BUTTON_MESSAGE =
```

Comment:
`EXPECTED_DELETE_BUTTON_MESSAGE` > `EXPECTED_DELETE_BUTTON_ERROR_MESSAGE`

Reply:
Well, we have `EXPECTED_SUCCESS_NOTIFICATION` and this message is not about possible success IMHO :-)
If there could be different messages near to "Delete" button, we need to name them differently according to content of message.
How about `PROJECTS_CANNOT_BE_REMOVED_IN_STOPPED_WORKSPACE_WARNING`?